### ISSUE TITLE
Send invitations asynchronously

### ISSUE THREAD

**Issue by Toniob** [nextcloud/server#46839]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

Hello. Currently, when you send invitations for a meeting in the Agenda app, the emails are sent when you validate the form. When you have several attendees, and you use an external SMTP server, the emails can take a few seconds to be sent. The users may have the feeling that something is going wrong. Maybe it’ll be better to send those invitation email asynchronously ?

If the synchronous method is to be kept, maybe showing the number of emails already sent, and the total number to send would be a good compromise ? Something like : Sending invitations : 2 / 5 …

**Comment by ChristophWurst** [comment]
Yes we could consider that. Nextcloud doesn't have an instant background worker queue at the moment and the process model of PHP otherwise doesn't allow for async tasks.
What we could do is send the invites in a background job. Ideally these are triggered every 5 minutes when the instance uses cron. But some instances still use the ancient 15 minute interval, or ajax backrgound jobs. Then the invite sending becomes delayed. This may make Nextcloud appear "slow".
